cleverhans/attacks.py

```diff
@@ -35,48 +35,12 @@ def fgsm(x, predictions, eps, back='tf', clip_min=None, clip_max=None):
     """
     if back == 'tf':
         # Compute FGSM using TensorFlow
+        from .attacks_tf import fgsm as fgsm_tf
         return fgsm_tf(x, predictions, eps, clip_min=clip_min, clip_max=clip_max)
     elif back == 'th':
-        raise NotImplementedError("Theano FGSM not implemented.")
-
-
-def fgsm_tf(x, predictions, eps, clip_min=None, clip_max=None):
-    """
-    TensorFlow implementation of the Fast Gradient
-    Sign method.
-    :param x: the input placeholder
-    :param predictions: the model's output tensor
-    :param eps: the epsilon (input variation parameter)
-    :param clip_min: optional parameter that can be used to set a minimum
-                    value for components of the example returned
-    :param clip_max: optional parameter that can be used to set a maximum
-                    value for components of the example returned
-    :return: a tensor for the adversarial example
-    """
-
-    # Compute loss
-    y = tf.to_float(tf.equal(predictions, tf.reduce_max(predictions, 1, keep_dims=True)))
-    y = y / tf.reduce_sum(y, 1, keep_dims=True)
-    loss = utils_tf.tf_model_loss(y, predictions, mean=False)
-
-    # Define gradient of loss wrt input
-    grad, = tf.gradients(loss, x)
-
-    # Take sign of gradient
-    signed_grad = tf.sign(grad)
-
-    # Multiply by constant epsilon
-    scaled_signed_grad = eps * signed_grad
-
-    # Add perturbation to original example to obtain adversarial example
-    adv_x = tf.stop_gradient(x + scaled_signed_grad)
-
-    # If clipping is needed, reset all values outside of [clip_min, clip_max]
-    if (clip_min is not None) and (clip_max is not None):
-        adv_x = tf.clip_by_value(adv_x, clip_min, clip_max)
-
-    return adv_x
-
+        # Compute FGSM using Theano
+        from .attack_th import fgsm as fgsm_th
+        raise fgsm_th(x, predictions, eps, clip_min=clip_min, clip_max=clip_max)
```

Comment:
This should be `return` rather than `raise`. Would you mind adding a small smoke test that would have failed for this?